app/models/user.rb

```diff
@@ -0,0 +1,48 @@
+require 'digest'
+
+class User < ApplicationRecord
+  enum :role, { admin: 0, buyer: 1, seller: 2, rider: 3 }
+
+  has_many :products, dependent: :destroy
+  has_many :reviews, dependent: :destroy
+  has_many :orders, dependent: :destroy
+  has_one :cart, dependent: :destroy
+
+  validates :username, presence: true, uniqueness: { case_sensitive: false }
+  validates :email, presence: true, uniqueness: { case_sensitive: false }, format: { with: URI::MailTo::EMAIL_REGEXP }
+
+  validates :phone, presence: true
+  validates :phone, format: {
+    with: /\A(017|013|018|019|015)\d{8}\z/,
+    message: 'must start with 0 and contain exactly 11 digits'
+  }
```

Comment:
may be we can introduce phony gem for phone number validations.

Reply:
Okay bhai. Adding it. Thank you.